routing/routing.go

```diff
@@ -152,22 +152,37 @@ type RouteFilter struct {
 
 // LBMetrics contains metrics used by LB algorithms
 type LBMetrics struct {
-	inflightRequests int64
+	endpointRegistryEntry Metrics
+}
+
+type LBMeticsOptions struct {
+	EndpointRegistryEntry Metrics
+}
+
+func NewLBMetrics(o *LBMeticsOptions) *LBMetrics {
+	return &LBMetrics{
+		endpointRegistryEntry: o.EndpointRegistryEntry,
+	}
 }
 
 // IncInflightRequest increments the number of outstanding requests from the proxy to a given backend.
 func (m *LBMetrics) IncInflightRequest() {
```

Comment:
I am wondering if we should return LBMetrics instead of Metrics interface  from EndpointRegistry
```go
func (r *EndpointRegistry) GetMetrics(key string) *LBMetrics
```

Reply:
That's possible, see https://github.com/zalando/skipper/pull/2841.
However, I stick to merging the PR 2823 as it is at the moment, because non-breaking change add too many boilerplate code: a lot of LBMetrics methods which essentially only call routing.Metrics methods:
https://github.com/zalando/skipper/pull/2841/files#diff-f58538a9de4e0a1e57e1b56ca2ed06fb8e8a149f68f0f5ea17503b3504ccae72R169-R190